queue/storage.go

```diff
@@ -0,0 +1,46 @@
+package queue
+
+import (
+	"context"
+	"sync"
+
+	"github.com/hashicorp/golang-lru/simplelru"
+)
+
+type Storage interface {
+	Exists(ctx context.Context, key string) bool
+}
+
+var _ Storage = (*localStorage)(nil)
```

Comment:
What does this line do? Pardon/Indulge my lack of Go knowledge

Reply:
> What does this line do? Pardon/Indulge my lack of Go knowledge

The `_` in front of the variable tells the compiler to effectively discard the RHS value, but to type-check it and evaluate it if it has any side effects, but the anonymous `Storage` variable per se doesn't take any process space. It's basically a way to ensure that localstorage implements type Storage at compile time.